### DISCUSSION TITLE
Best ways to speed up a training loop?

### DISCUSSION THREAD

**Discussion by pvasired** [jax-ml/jax#14160] [Q&A]
Hi,

I have a training loop which is time limited by `jax.grad()` calls, and I am wondering what are the most common ways of speeding up JAX? I have heard there may be some speedups from different datatypes (i.e. bfloat16), but I wanted to know if there are some "freebies" I could implement to calculate gradients faster.

The function I am differentiating is already vectorized, and I'm just using the default data type.
